subxt/src/rpc/lightclient/mod.rs

```diff
@@ -0,0 +1,241 @@
+pub enum LightClientError {
+    /// Error encountered while adding the chain to the light-client.
+    #[error("Failed to add the chain to the light client: {0}.")]
+    AddChainError(String),
+    /// The background task is closed.
+    #[error("Failed to communicate with the background task.")]
+    BackgroundClosed,
+    /// Invalid RPC parameters cannot be serialized as JSON string.
+    #[error("RPC parameters cannot be serialized as JSON string.")]
+    InvalidParams,
+    /// Error originated while trying to submit a RPC request.
+    #[error("RPC request cannot be sent: {0}.")]
+    Request(String),
+    /// The provided URL scheme is invalid.
+    ///
+    /// Supported versions: WS, WSS.
+    #[error("The provided URL scheme is invalid.")]
+    InvalidScheme,
+    /// The provided URL is invalid.
+    #[error("The provided URL scheme is invalid.")]
+    InvalidUrl,
+    /// Handshake error while connecting to a node.
+    #[error("WS handshake failed.")]
+    Handshake,
+    /// The configuration provided to the builder is compatible.
+    ///
+    /// The light client can be constructed with either a chain config
+    /// or an URL to a trusted node.
+    #[error("The configuration provided to the builder is compatible. Please provide a chain spec or URL.")]
+    IncompatibleConfig,
+}
+
+/// Builder for [`LightClient`].
+#[derive(Clone, Debug)]
+pub struct LightClientBuilder<'a> {
+    chain_spec: Option<&'a str>,
+    max_pending_requests: NonZeroU32,
+    max_subscriptions: u32,
+    database_content: Option<&'a str>,
+    url: Option<&'a str>,
+    bootnodes: Option<Vec<String>>,
+    potential_relay_chains: Option<Vec<ChainId>>,
+}
+
+impl<'a> Default for LightClientBuilder<'a> {
+    fn default() -> Self {
+        Self {
+            chain_spec: None,
+            max_pending_requests: NonZeroU32::new(128)
+                .expect("Valid number is greater than zero; qed"),
+            max_subscriptions: 1024,
+            database_content: None,
+            url: None,
+            bootnodes: None,
+            potential_relay_chains: None,
+        }
+    }
+}
+
+impl<'a> LightClientBuilder<'a> {
+    /// Create a new [`LightClientBuilder`].
+    pub fn new() -> LightClientBuilder<'a> {
+        LightClientBuilder::default()
+    }
+
+    /// The most important field of the configuration is the chain specification.
+    /// This is a JSON document containing all the information necessary for the client to
+    /// connect to said chain.
+    ///
+    /// The chain spec must be obtained from a trusted entity.
+    ///
+    /// It can be fetched from a trused node with the following command:
+    /// ```bash
+    /// curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "sync_state_genSyncSpec", "params":[true]}' http://localhost:9944/ | jq .result > res.spec
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// For testing environments, please populate the "bootNodes" if the not already provided.
```

Comment:
Or use `bootnodes()` on this builder thing to configure some? (which might be easier than editing the spec, so maybe is a better way to push people?)